File: chess-game/chess-game.v3/web/client/chess-app/src/app/pages/game/play/play/models/pieces/queen.ts
import { Cell } from "../board/ cell";
import { Side } from "../side";
import { Piece } from "./piece";
import { PieceType } from "./piece_type";

export class Queen extends Piece {
    
    override get can_move_over(): boolean {
        return false;
    }
    
    constructor(piece_id: string, side: Side) {
        super(piece_id, side, PieceType.Queen);
    }

    override validateMove(from: Cell, to: Cell): boolean {
        const [delta_file, delta_rank] = this.calculateMoveDeltas(from, to);
        // rook
        if (delta_file == 0 || delta_rank == 0) {
            return true;
        }

        // bishop
        return delta_file == delta_rank
    }
}

File: backend/chessapp/domain/pieces/queen.py
from ...domain.pieces.piece import Piece
from ...domain.pieces.piece_type import PieceType
from ...domain.value_objects.side import Side

class Queen(Piece):

    def get_points(self):
        return 90

    def __init__(self, side: Side):
        super().__init__(side, PieceType.Queen)